### PR TITLE
feat(config): add field and value matchers to the commit parser

### PR DESCRIPTION
**Pull request by dpecos**
## Description

This PR introduces the capability of grouping commits by attributes of the commit, i.e. `author.name` or `author.email`. 

## Motivation and Context

This changeset should cover the scenario described in #194.

The reporter of that issue wanted to create a new group based gathering the commits for a specific author name:

```
 { author.name = "Renovate Bot", group = "Dependency Updates"},
```
## How Has This Been Tested?

In order to test it works, I've modified the `cliff.toml` file in this project to include a new group like this:

```
commit_parsers = [
  { field = "author.name", pattern="dependabot\\[bot\\]", group = "Dependency Updates"},
  ...
```

which generates the following new group:

```
### Dependency Updates

- *(deps)* Bump regex from 1.9.6 to 1.10.0 ([#308](https://github.com/orhun/git-cliff/issues/308)) - ([a6a890a](https://github.com/orhun/git-cliff/commit/a6a890ac4f0051518b4b4c0d65a5ce83ae0de1c8))
- *(deps)* Bump toml from 0.8.1 to 0.8.2 ([#306](https://github.com/orhun/git-cliff/issues/306)) - ([8422096](https://github.com/orhun/git-cliff/commit/8422096549320604643cda9baab4190a5de272fb))
- *(deps)* Bump regex from 1.9.5 to 1.9.6 ([#305](https://github.com/orhun/git-cliff/issues/305)) - ([e263163](https://github.com/orhun/git-cliff/commit/e2631639bf0438fb8f67e3b99c27354fd7409f57))
```
## Screenshots / Logs (if applicable)

image

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: git-cliff-core/src/commit.rs [hunk 1]
```diff
@@ -258,6 +258,67 @@ impl Commit<'_> {
...
+					)),
+					_ => {
+						return Err(AppError::FieldError(format!(
+							"field does not exist: {}",
+							field_name
+						)))
+					}
+				}
+			}
+
```
Comment (orhun) [inline git-cliff-core/src/commit.rs:321]: Let's simplify this block as:

```suggestion
			if let (Some(field_name), Some(pattern_regex)) =
				(parser.field.as_ref(), parser.pattern.as_ref())
			{
				regex_checks.push((
					pattern_regex,
					match field_name.as_str() {
						"id" => Some(self.id.clone()),
						"message" => Some(self.message.clone()),
						"body" => self
							.conv
							.as_ref()
							.and_then(|v| v.body())
							.map(|v| v.to_string()),
						"author.name" => self.author.name.clone(),
						"author.email" => self.author.email.clone(),
						"committer.name" => self.committer.name.clone(),
						"committer.email" => self.committer.email.clone(),
						_ => None,
					}
					.ok_or_else(|| {
						AppError::FieldError(format!(
							"field {} does not have a value",
							field_name
						))
					})?,
				));
			}
```

**Review by orhun — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution! Just had some suggestions for improving the code.

**Comment by orhun** [comment]
Also, can you resolve the conflicts with `main`?

**Comment by dpecos** [comment]
Suggestions applied (btw, I really like the refactor for the big switch case I had) and changes from main pulled :-)

**Review by orhun — APPROVED** [review]
LGTM, thanks for your contribution!

It was really fun working with you at the EuroRust workshop @dpecos! Wish to see you in the future events/open source projects. ❤️

**Pull request merged**